src/tools/kata-ctl/src/check.rs

```diff
@@ -197,6 +208,110 @@ pub fn check_official_releases() -> Result<()> {
     Ok(())
 }
 
+#[cfg(any(target_arch = "x86_64"))]
+pub fn check_kernel_module_loaded(module: &KernelModule) -> Result<()> {
+    const MODPROBE_PARAMETERS_DRY_RUN: &str = "--dry-run";
+    const MODPROBE_PARAMETERS_FIRST_TIME: &str = "--first-time";
+
+    // Partial check w/ modinfo
+    let output_modinfo = Command::new(MODINFO_PATH)
+        .arg(module.name.clone())
+        .stdout(Stdio::piped())
+        .output()?;
+
+    let status_modinfo = output_modinfo.status;
+
+    // Partial check w/ modprobe
+    let output_modprobe = Command::new(MODPROBE_PATH)
+        .arg(MODPROBE_PARAMETERS_DRY_RUN)
+        .arg(MODPROBE_PARAMETERS_FIRST_TIME)
+        .arg(module.name.clone())
+        .stdout(Stdio::piped())
+        .output()?;
+
+    let status_modprobe = output_modprobe.status;
+
+    // Complete check
+    if status_modinfo.success() && !status_modprobe.success() {
+        return Ok(());
+    }
+
+    match status_modprobe.code() {
+        Some(code) => {
+            // Module not found
+            if code != 0 {
+                let std_err = String::from_utf8_lossy(&output_modinfo.stderr);
+                let msg = format!("{} {}", ERR_LSMOD, std_err);
+                return Err(anyhow!(msg));
+            }
+
+            // module not loaded
+            let msg = format!("{} {}", ERR_MODPROBE, module.name);
+            Err(anyhow!(msg))
+        }
+        None => Err(anyhow!("Process terminated by signal")),
+    }
+}
+
+pub fn check_kernel_module_parameters(
+    module: &KernelModule,
+) -> std::result::Result<std::result::Result<std::string::String, anyhow::Error>, ()> {
+    const MODULES_PATH: &str = "/sys/module";
+    const CAT_PATH: &str = "/usr/bin/cat";
```

Comment:
We need to minimise use of any external commands. `modprobe` can't really be avoided but can you remove the use of `cat` and use https://doc.rust-lang.org/std/fs/fn.read_to_string.html instead? That would simplify the logic too I think.

Reply:
@jodh-intel fixed, thank you!